src/v/cluster/topic_table_probe.h

```diff
@@ -24,10 +24,23 @@ class topic_table_probe {
     void handle_topic_creation(create_topic_cmd::key_t);
     void handle_topic_deletion(const delete_topic_cmd::key_t&);
 
+    void handle_update_addition(const model::ntp& ntp);
+    void handle_update_removal(const model::ntp& ntp);
+
 private:
+    void setup_metrics();
+
+    void increment_moving_to_partitions() { moving_to_partitions += 1; }
+    void increment_moving_from_partitions() { moving_from_partitions += 1; }
+    void decrement_moving_to_partitions() { moving_to_partitions -= 1; }
+    void decrement_moving_from_partitions() { moving_from_partitions -= 1; }
+
     const topic_table& _topic_table;
     absl::flat_hash_map<model::topic_namespace, ss::metrics::metric_groups>
       _topics_metrics;
+    ss::metrics::metric_groups _metrics;
```

Comment:
This adds the metrics to the `/metrics` endpoint. We've not had a discussion about where new metrics should be exposed, but in this case I feel like these metrics should be exposed on both: `/metrics` and `/public_metrics`.

Generally, if the metric is not high cardinality (partition labels are something to look out for) and it could be useful to an operator without implementation knowledge, I think it belongs in `/public_metrics`.

Reply:
If you decide to add it, you'd need to create a new metrics group and add to both of them.
`sm::metric_groups _public_metrics{ssx::metrics::public_metrics_handle};`

Reply:
switched to public metrics